File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/DefaultAlerts.java
package closer.vlllage.com.closer.handler.helpers;

import android.support.annotation.StringRes;
import android.widget.TextView;

import closer.vlllage.com.closer.R;
import closer.vlllage.com.closer.pool.PoolMember;

public class DefaultAlerts extends PoolMember {

    public void thatDidntWork() {
        thatDidntWork(null);
    }

    public void thatDidntWork(String message) {
        $(AlertHandler.class).make()
                .setTitle($(ResourcesHandler.class).getResources().getString(R.string.that_didnt_work))
                .setMessage(message)
                .setPositiveButton($(ResourcesHandler.class).getResources().getString(R.string.ok))
                .show();
    }

    public void syncError() {
        $(AlertHandler.class).make()
                .setTitle($(ResourcesHandler.class).getResources().getString(R.string.sync_didnt_work))
                .setPositiveButton($(ResourcesHandler.class).getResources().getString(R.string.boo))
                .show();
    }

    public void longMessage(@StringRes Integer title, @StringRes int message) {
        longMessage(title, message);
    }

    public void longMessage(@StringRes Integer title, CharSequence message) {
        $(AlertHandler.class).make()
                .setTitle(title == null ? null : $(ResourcesHandler.class).getResources().getString(title))
                .setLayoutResId(R.layout.long_message_modal)
                .setOnAfterViewCreated((alertConfig, view) -> {
                    ((TextView) view.findViewById(R.id.messageText)).setText(message);
                })
                .setPositiveButton($(ResourcesHandler.class).getResources().getString(R.string.ok))
                .show();
    }

    public void message(String message) {
        $(AlertHandler.class).make()
                .setMessage(message)
                .setPositiveButton($(ResourcesHandler.class).getResources().getString(R.string.ok))
                .show();
    }

    public void message(String message, AlertConfig.ButtonCallback buttonCallback) {
        $(AlertHandler.class).make()
                .setMessage(message)
                .setPositiveButton($(ResourcesHandler.class).getResources().getString(R.string.ok))
                .setPositiveButtonCallback(buttonCallback)
                .show();
    }

    public void message(String title, String message) {
        $(AlertHandler.class).make()
                .setTitle(title)
                .setMessage(message)
                .setPositiveButton($(ResourcesHandler.class).getResources().getString(R.string.ok))
                .show();
    }

    public void message(@StringRes int titleRes, @StringRes int messageRes) {
        message($(ResourcesHandler.class).getResources().getString(titleRes),
                $(ResourcesHandler.class).getResources().getString(messageRes));
    }

    public void message(@StringRes int stringRes) {
        message($(ResourcesHandler.class).getResources().getString(stringRes));
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/DefaultAlerts.kt
package closer.vlllage.com.closer.handler.helpers

import androidx.annotation.StringRes
import android.view.View
import android.widget.TextView

import closer.vlllage.com.closer.R
import closer.vlllage.com.closer.databinding.LongMessageModalBinding
import com.queatz.on.On

class DefaultAlerts constructor(private val on: On) {

    fun thatDidntWork(message: String? = null) {
        on<AlertHandler>().make().apply {
                    title = on<ResourcesHandler>().resources.getString(R.string.that_didnt_work)
                    this.message = message
                    positiveButton = on<ResourcesHandler>().resources.getString(R.string.ok)
                    show()
                }
    }

    fun syncError() {
        on<AlertHandler>().make().apply {
            title = on<ResourcesHandler>().resources.getString(R.string.sync_didnt_work)
            positiveButton = on<ResourcesHandler>().resources.getString(R.string.boo)
            show()
        }
    }

    fun longMessage(@StringRes title: Int?, @StringRes message: Int) {
        longMessage(title, message)
    }

    fun longMessage(@StringRes title: Int?, message: CharSequence) {
        on<AlertHandler>().view { LongMessageModalBinding.inflate(it) }.apply {
            this.title = if (title == null) null else on<ResourcesHandler>().resources.getString(title)
            onAfterViewCreated = { _, view -> view.messageText.text = message }
            positiveButton = on<ResourcesHandler>().resources.getString(R.string.ok)
            show()
        }
    }

    fun message(message: String) {
        on<AlertHandler>().make().apply {
            this.message = message
            positiveButton = on<ResourcesHandler>().resources.getString(R.string.ok)
            show()
        }

    }

    fun message(message: String, buttonCallback: (Any?) -> Unit) {
        on<AlertHandler>().make().apply {
            this.message = message
            positiveButton = on<ResourcesHandler>().resources.getString(R.string.ok)
            positiveButtonCallback = buttonCallback
            show()
        }
    }

    fun message(title: String, message: String, closeCallback: (() -> Unit)? = null) {
        on<AlertHandler>().make().apply {
            this.title = title
            this.message = message
            positiveButton = on<ResourcesHandler>().resources.getString(R.string.ok)
            positiveButtonCallback = { closeCallback?.invoke() }
            negativeButtonCallback = { closeCallback?.invoke() }
            cancelIsNegative = true
            show()
        }
    }

    fun message(@StringRes titleRes: Int, @StringRes messageRes: Int) {
        message(on<ResourcesHandler>().resources.getString(titleRes),
                on<ResourcesHandler>().resources.getString(messageRes))
    }

    fun message(@StringRes stringRes: Int) {
        message(on<ResourcesHandler>().resources.getString(stringRes))
    }
}
